feat(test-files): Convert test files to vitest

jurriaanr opened the pull request:
I left the imports in there, adding them to common import would only polute the import namespace more and the current solution works.

Output is.. okayish, if you call a million "ESLint: Unsafe call of a(n) `error` type typed value.(@typescript-eslint/ no-unsafe-call)" okayish... some references to t and createTestEvent are still in there, no doubt manual work is still necessary. Todo of replacing t.pass() with something better is still open, mainly because I didin't really find a better alternative. Also replacing the teardown functions is still a nice to have, while finding them is easy, replacing them may be quute hard to do.

Commit:
init

Commit:
init

Commit:
Merge branch 'main' into convert-test-files

# Conflicts:
#	packages/compas-convert/src/index.ts
#	packages/compas-convert/src/passes/add-common-imports.ts
#	packages/compas-convert/src/passes/init-typescript-in-project.ts
#	packages/compas-convert/src/shared/jsdoc.ts

Commit:
parse innerblock items by going over all descendants

Commit:
added todo's

jurriaanr committed:
Update convert-test-files.ts

Signed-off-by: jurriaanr <jurriaanr@users.noreply.github.com>

jurriaanr committed:
Update convert-test-files.ts

Signed-off-by: jurriaanr <jurriaanr@users.noreply.github.com>

jurriaanr committed:
Update convert-test-files.ts

Signed-off-by: Jurriaan Ruitenberg <jurriaan@lightbase.nl>

jurriaanr committed:
Update convert-test-files.ts

Signed-off-by: jurriaanr <jurriaanr@users.noreply.github.com>

jurriaanr committed:
Update convert-test-files.ts

Signed-off-by: Jurriaan Ruitenberg <jurriaan@lightbase.nl>

Commit:
first go

Commit:
first go

Signed-off-by: Jurriaan Ruitenberg <jurriaan@lightbase.nl>

Commit:
merge

Commit:
index restored

Commit:
index restored

Signed-off-by: Jurriaan Ruitenberg <jurriaan@lightbase.nl>

Commit:
removed logging

Commit:
removed logging

Signed-off-by: Jurriaan Ruitenberg <jurriaan@lightbase.nl>

Commit:
linting stuff

Commit:
linting stuff

Signed-off-by: Jurriaan Ruitenberg <jurriaan@lightbase.nl>

dirkdev98 reviewed (commented):
> Output is.. okayish, if you call a million "ESLint: Unsafe call of a(n) error type typed value.(@typescript-eslint/ no-unsafe-call)" okayish

Adding `vitest` as a dependency in `initTypescriptInProject` should fix a bunch of these errors.

When you are there, might as well add `scripts["test"] = "vitest"` in there :)

---

Can we add the required vitest config to call the global setup & teardown

---

> Todo of replacing t.pass() with something better is still open, mainly because I didin't really find a better alternative.

As discussed, no good alternative. For backwards compat we should use https://vitest.dev/config/#expect-requireassertions in the config. Since a bunch of test-cases expect that the runner fails when no assertions have been executed.

---

Lastly, can you either fix the commiter email of all commits / add a sign-off as requested by the DCO?

Commit:
Merge remote-tracking branch 'refs/remotes/origin/convert-test-files' into convert-test-files

Commit:
added import for newTestEvent

Signed-off-by: Jurriaan Ruitenberg <jur@lightbase.nl>

Commit:
DCO Remediation Commit for Jur <jur@oberon.nl>

I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 8dcf4de2497bf460098bbfae93093702b7d23b06
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: de370fa97603b83c7072ce063ab157b1867c5380
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 9df2c742fa4ab2da7dd90cdd631c4189f388f169
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: cc9c5828dec3ba32794e6b4fdd6eae1d23415ead
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 3c589f51e78587618bb731126f8b5b6f02034cb1
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 1500d7f252eb9532584d7caa9be138b65cb51599
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 0aec64dcd39c34b877d2a0e19dc01c5ec231c6dd
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 17cd36dd1d0fe97e4608779caae157b97d26a43d
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 63d44249215d57212e0834b9c433880a6967edfd
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 96770531d5ae9c4ba7651de8e216ff84a76ac15b
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 959a03de071c71dc4441b7ef62c08c54116723e0
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 604fd6d494498916a2acd0a0b8d73b354bb21266

Signed-off-by: Jur <jur@oberon.nl>

Commit:
DCO Remediation Commit for Jur <jur@oberon.nl>

I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 8dcf4de2497bf460098bbfae93093702b7d23b06
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: de370fa97603b83c7072ce063ab157b1867c5380
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 9df2c742fa4ab2da7dd90cdd631c4189f388f169
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: cc9c5828dec3ba32794e6b4fdd6eae1d23415ead
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 3c589f51e78587618bb731126f8b5b6f02034cb1
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 1500d7f252eb9532584d7caa9be138b65cb51599
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 0aec64dcd39c34b877d2a0e19dc01c5ec231c6dd
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 17cd36dd1d0fe97e4608779caae157b97d26a43d
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 63d44249215d57212e0834b9c433880a6967edfd
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 96770531d5ae9c4ba7651de8e216ff84a76ac15b
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 959a03de071c71dc4441b7ef62c08c54116723e0
I, Jur <jur@oberon.nl>, hereby add my Signed-off-by to this commit: 604fd6d494498916a2acd0a0b8d73b354bb21266

Signed-off-by: Jur <jur@oberon.nl>

jurriaanr committed:
DCO Remediation Commit for jurriaanr <jurriaanr@users.noreply.github.com>

I, jurriaanr <jurriaanr@users.noreply.github.com>, hereby add my Signed-off-by to this commit: e161487a94d785a300ee7bff75df790f75b98439
I, jurriaanr <jurriaanr@users.noreply.github.com>, hereby add my Signed-off-by to this commit: 57eb8ea79ab02e2a2681b63abb524039af986946

Signed-off-by: jurriaanr <jurriaanr@users.noreply.github.com>

Commit:
DCO Remediation Commit for jurriaanr <jurriaanr@users.noreply.github.com>

I, jurriaanr <jurriaanr@users.noreply.github.com>, hereby add my Signed-off-by to this commit: e161487a94d785a300ee7bff75df790f75b98439
I, jurriaanr <jurriaanr@users.noreply.github.com>, hereby add my Signed-off-by to this commit: 57eb8ea79ab02e2a2681b63abb524039af986946

Signed-off-by: jurriaanr <jurriaanr@users.noreply.github.com>

Commit:
DCO Remediation Commit for Jurriaan Ruitenberg <jur@lightbase.nl>

I, Jurriaan Ruitenberg <jur@lightbase.nl>, hereby add my Signed-off-by to this commit: d6ae806525d03222563910043f4160d8d89cf24d
I, Jurriaan Ruitenberg <jur@lightbase.nl>, hereby add my Signed-off-by to this commit: 326e96aad970e08fadef2aafb629469622d8c33e

Signed-off-by: Jurriaan Ruitenberg <jur@lightbase.nl>
